marda_extractors_api/__init__.py

```diff
@@ -48,6 +48,8 @@ def extract(
     preferred_mode: SupportedExecutionMethod | str = SupportedExecutionMethod.PYTHON,
     install: bool = True,
     use_venv: bool = True,
+    extractor_definition: dict | None = None,
```

Comment:
When was this syntax introduced? Is the required python version consistent with this syntax?

Reply:
3.10 so I think we're on solid ground; this does enforce 3.10 support on the extractors too though -- we could consider checking the version numbers and using the (yet to be developed) docker integration in cases where the Python versions are incompatible

Reply:
(https://peps.python.org/pep-0604/)

Reply:
The problem is that the venvs made by the code in this repo will have the same version as the env its running in. Not trivial to create venvs of another version without also depending on an external tool (e.g. pyenv). That's a separate problem to this though.